include/ofi_mr.h

```diff
@@ -111,6 +116,23 @@ struct ofi_mem_monitor {
 			 const void *addr, size_t len);
 	void (*unsubscribe)(struct ofi_mem_monitor *notifier,
 			    const void *addr, size_t len);
+
+	/* Get HMEM information from the memory monitor. If a memory monitor
+	 * does not support HMEM information, -FI_ENOSYS is returned. -FI_ENOSYS
+	 * should not be treated as an error.
+	 */
+	int (*get_hmem_info)(struct ofi_mem_monitor *notifier,
+			     const void *addr, size_t len,
+			     union ofi_mr_hmem_info *hmem_info);
```

Comment:
If the monitor doesn't need the hmem info, it should just return success, so we don't have 2 non-error return values to check.  The patch does not make the purpose of this call clear.  The hmem_info is part of ofi_mr_entry, so we have it once we find the entry.  We just need to pass it into the valid callback.

I expected to see this passed into the subscribe and unsubscribe calls, so that it can be initialized as part of the subscription process.

Reply:
Ok. I'll remove get_hmem_info() and have ofi_mr_hmem_info initialized during subscription.